src/psyclone/dynamo0p3.py

```diff
@@ -5014,37 +5028,43 @@ def __init__(self, alg_invocation, idx, invokes):
                     global_sum = DynGlobalSum(scalar, parent=loop.parent)
                     loop.parent.children.insert(loop.position+1, global_sum)
 
-    def unique_proxy_declarations(self, datatype, access=None):
-        ''' Returns a list of all required proxy declarations for the
-        specified datatype.  If access is supplied (e.g. "AccessType.WRITE")
+    def unique_proxy_declarations(self, argument_type, access=None):
+        '''
+        Returns a list of all required proxy declarations for the specified
+        argument type. If access is supplied (e.g. "AccessType.WRITE")
         then only declarations with that access are returned.
-        :param str datatype: Datatype that proxy declarations are \
-                             searched for.
-        :param access: optional AccessType for the specified data type.
+
+        :param str argument_type: argument type that proxy declarations are \
+                                  searched for.
+        :param access: optional AccessType for the specified argument type.
         :type access: :py:class:`psyclone.core.access_type.AccessType`.
-        :return: a list of all required proxy declarations for the \
-                 specified datatype.
-        :raises GenerationError: if datatype is invalid.
+
+        :returns: a list of all required proxy declarations for the \
+                  specified argument type.
+        :rtype: list of str
+
+        :raises GenerationError: if the supplied argument type is invalid.
         :raises InternalError: if an invalid access is specified, i.e. \
-                not of type AccessType.
+                               not of type AccessType.
+
         '''
-        if datatype not in GH_VALID_ARG_TYPE_NAMES:
+        if argument_type not in GH_VALID_ARG_TYPE_NAMES:
             raise GenerationError(
```

Comment:
Not your fault but since we're here: I think this should be an `InternalError` as it 'should never happen'.

Reply:
Agreed. Changed.